docs/source/dev/tests.md

```diff
@@ -0,0 +1,94 @@
+
+![e2e](assets/e2eTest.png)
+
+The default browser for this process is chosen to be _Firefox_.
+
+Another possibility is to use the cypress client to run the tests. This gives you some additional possibilities such as choosing the browser environment, run only specific specs or using an interactive debugging mode. To run the cypress client, you can run the following command:
+
+```
+# Make sure to run this from the root of the project
+cd mxcubeweb
+
+pnpm --prefix ui run cypress
+```
+
+![cypress_client](assets/cypress_client.png)
+
+### Best Practices
+
+The best use for development would be to test the components you changed first individually by using the cypress client. This allows for better error tracking and time efficiency as it prevents you from running tests on components that are not effected by your changes multiple times during development.
+
+Once all specs concerning the changed components passed the tests, please make sure to pass every test in the headless browser environment at least once **before** you push your changes. (_Github Actions_ will run this command during a pull request and will block your changes from merging if they fail the tests)
+
+How to run the cypress client and the headless browser environment is explained in the previous section.
+
+**Only** if necessary make changes to existing tests. In case, your changes include the conceptual change on how users might perform certain actions or the changes of element name/inner texts you are allowed to change the tests. (Please make sure, that these changes are reasonable)
+
+### Troubleshooting
+
+This section includes a few possible scenarios on why your tests may be failing, please read through them, as this might save you some valuable time.
+
+The first thing you should do, when failing a test is to re-run it. This solution might sound simple but failing a test case, does not necessary mean, that your code is wrong and other factors might play a role in it.
+
+When the re-run did not help, try to switch the testing browser, you can do so by using the cypress client. However, please make sure that your code works with most browsers, especially with chrome and firefox. 
+
+It is important to notice that cypress will not test in full-screen mode, please make sure, that buttons and other elements stay visible and clickable in smaller test-windows.
+
+The cypress tests are configured to act according to human behavior, hence during testing, certain button names or texts are looked instead of element attributes. If for a reasonable argument, your code adjustments change these names, please make sure to change the tests accordingly.
+
+**Very important**: If you are using the development server while simultaneously running tests, do not take control (for mounting samples, running queue, ...) on the development server. This might result in tests failing, because they needed to access a certain resource, that is not accessible in _Observer mode_.
+
+### Contributing
```

Comment:
We already have a contributing document.

So I guess we should link the two documents back and forth.

Reply:
I agree, that is a good idea

Reply:
I'm wondering, should we move the contents of `CONTRIBUTING.md` so that its only in the documentation and simply only have a link pointing to the part in `CONTRIBUTING.md`. I know that `CONTRIBUTING.md` is automatically included but it can be a bit confusing.

Reply:
@marcus-oscarsson Yes, we can consider that. I created a ticket: https://github.com/mxcube/mxcubecore/issues/955

Reply:
Then @walesch-yan let's consider this conversation resolved. We will handle the links back and forth separately.